src/components/Application-Body/JobsAppBody.tsx

```diff
@@ -31,40 +32,43 @@ const styles = makeStyles({
     }
 });
 
-const JobsAppBody: React.FC<{}> = (): JSX.Element => {
-  const classes = styles({});
-  const [ads, setAds] = useState<Advertisement[]>([]);
-  const [searchValue, setSearchValue] = useState<string>('');
+interface JobsAppBodyOwnProps {
+    jobsStore?: JobsStore;
+}
 
-  const fetchAllAds = (): void => {
-    getAllAds().then(data => setAds(data));
-  }
+const JobsAppBody: React.FC<JobsAppBodyOwnProps> = (props): JSX.Element => {
+    const classes = styles({});
+    const jobsStore = props.jobsStore!;
 
-  useEffect(() => {
-      fetchAllAds();
-  }, []);
+    const [searchValue, setSearchValue] = useState<string>('');
+
+    useEffect(() => {
+        jobsStore.loadAdvertisements();
+    }, []);
 
     const onSearchValueChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
         setSearchValue(event.target.value);
     }
 
     const getFilteredAds = (): Advertisement[] => {
+        const ads = jobsStore.advertisements;
+
         return ads.filter(ad => (
                 ad.name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1 ||
                 ad.tag.name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
             )
         );
     }
 
-  return (
-    <div className={classes.appBodyContent}>
-        <Header 
-            searchValue={searchValue} 
-            onSearchValueChange={onSearchValueChange}
-            fetchAllAdsAfterPost={fetchAllAds} />
-        <JobsList ads={getFilteredAds()} isFiltered={searchValue !== '' && ads.length !== 0}/>
-    </div>
-  );
+    return (
+        <div className={classes.appBodyContent}>
+            <Header 
+                searchValue={searchValue} 
+                onSearchValueChange={onSearchValueChange}
+                fetchAllAdsAfterPost={() => {jobsStore.loadAdvertisements()}} />

```

Comment:
Refererence to method doesn't work?

fetchAllAdsAfterPost={jobsStore.loadAdvertisements}